### PR TITLE
Feature/5706 add form 1 efile endpoint

### LINKED ISSUE #5706: Consider adding efile/form1
### What we’re after

We are adding efile/form2 so that our users have real time access to our form 2 data. We should consider doing this for our form 1 data if we could support it performance-wise. 
 

### Action item(s)

(These are the smaller tasks that should happen in order to complete this work)
- [x] Research if this would be helpful to users --what fields are missing from efilings
- [x] Build out efile/form1 
- [x] Load test

### Completion criteria

(What does the end state look like - as long as this task(s) is done, this work is complete)
- [ ]

### PR DESCRIPTION
**Pull request by tmpayton**
## Summary (required)

- Resolves #5706 

This ticket creates a new efile/form1 endpoint. I included fields for candidate, committee, affiliated committees, and treasurer.  [Here](https://docs.google.com/spreadsheets/d/1Y1cElQDiAKK3SoXxPvg3ItV2NzWAGZz8G8z7AOFFndA/edit#gid=1782497329) is a link to the load testing results.

## Impacted areas of the application

General components of the application that this PR will affect:

- efiling endpoint

## How to test

- Checkout this branch 
- `flask run`
- `pytest`
- test efile/form1 endpoint and filters

Sample URLs: 

http://127.0.0.1:5000/v1/efile/form1/?page=1&per_page=20&sort=-load_timestamp&sort_hide_null=false&sort_null_only=false&sort_nulls_last=false

http://127.0.0.1:5000/v1/efile/form1/?page=1&per_page=20&committee_type=H&sort=-load_timestamp&sort_hide_null=false&sort_null_only=false&sort_nulls_last=false

http://127.0.0.1:5000/v1/efile/form1/?page=1&per_page=20&election_state=NV&sort=-load_timestamp&sort_hide_null=false&sort_null_only=false&sort_nulls_last=false

http://127.0.0.1:5000/v1/efile/form1/?page=1&per_page=20&candidate_office=P&sort=-load_timestamp&sort_hide_null=false&sort_null_only=false&sort_nulls_last=false

http://127.0.0.1:5000/v1/efile/form1/?page=1&per_page=20&min_load_timestamp=03%2F27%2F2024&sort=-load_timestamp&sort_hide_null=false&sort_null_only=false&sort_nulls_last=false

### CODE REVIEW

**Review by fec-jli — APPROVED** [review]
It looks good. Thanks

File: tests/test_filings.py [hunk 1]
```diff
@@ -546,3 +546,167 @@ def test_filter_election_state(self):
...
+        factories.Form1EFilingFactory(candidate_id="H00000001")
+        factories.Form1EFilingFactory(candidate_id="S00000002")
+        factories.Form1EFilingFactory(committee_id="C00000002")
+
+        results = self._results(api.url_for(F1EFilingsView))
+        self.assertEqual(len(results), 3)
+
+    def test_filter_load_timestamp(self):
+        factories.Form1EFilingFactory(
+            candidate_id="H010",
```
[the remark this answers is not in the document]
Comment (tmpayton) [reply tests/test_filings.py]: > @tmpayton In test cases, candiate_Id length should be 9, right?

@pkfec Fixed! Thank you for the review.

**Comment by pkfec** [comment]
@tmpayton on efile/form 1 endpoint can you add field validations on candidate_id, committee_id and file_number? This endpoint returns 200 when the input is incorrect

Examples: 
committee_id=C
http://127.0.0.1:5000/v1/efile/form1/?page=1&per_page=20&committee_id=C&sort=-load_timestamp&sort_hide_null=false&sort_null_only=false&sort_nulls_last=false

file_number=12:
http://127.0.0.1:5000/v1/efile/form1/?page=1&per_page=20&file_number=123&sort=-load_timestamp&sort_hide_null=false&sort_null_only=false&sort_nulls_last=false

**Comment by tmpayton** [comment]
> @tmpayton on efile/form 1 endpoint can you add field validations on candidate_id, committee_id and file_number? This endpoint returns 200 when the input is incorrect.
> 
> Examples: committee_id=C http://127.0.0.1:5000/v1/efile/form1/?page=1&per_page=20&committee_id=C&sort=-load_timestamp&sort_hide_null=false&sort_null_only=false&sort_nulls_last=false
> 
> file_number=12: http://127.0.0.1:5000/v1/efile/form1/?page=1&per_page=20&file_number=123&sort=-load_timestamp&sort_hide_null=false&sort_null_only=false&sort_nulls_last=false

@pkfec I validate that file number is an integer greater than zero. I'm not sure exactly how big or small a file number could potentially be. There is no cap in the database.

**Review by pkfec — APPROVED** [review]
Changes look good, thanks @tmpayton

**Review by cnlucas — APPROVED** [review]
Great job, @tmpayton, thank you!

**Pull request merged**